firedrake/mg/embedded.py

```diff
@@ -93,14 +93,17 @@ def cache(self, V):
         except KeyError:
             return self.caches.setdefault(key, TransferManager.Cache(*key))
 
+    def cache_key(self, V):
+        return (V.mesh(), tuple(V.boundary_set))
```

Comment:
Why do we need to include `V.mesh()` in the key? Can we include `V.boundary_set` in the key in the `cache` method just in the above?

Reply:
Please cache *on* the mesh. And this can be another PR.
And use `frozenset(V.boundary_set)` instead of `tuple(...)`.